src/Linters/BaseASTLinter.php

```diff
@@ -76,13 +77,28 @@ public function getPrettyTextForNode(
     foreach ($this->ast->traverseWithParents() as list($node, $parents)) {
       if ($node instanceof $target) {
         $error = $this->getLintErrorForNode($node, $parents);
-        if ($error !== null) {
+
+        if ($error !== null && !$this->isLinterDisabled($node, $error)) {
           yield $error;
         }
       }
     }
   }
 
+  /**
+   * Allow users to disable either all linter errors of this type in a file or
+   * specific cases where a linter is used.
+   **/
+  protected function isLinterDisabled(EditableNode $node, LintError $error): bool {
+    // Is this specific instance of the linter error disabled?
+    $token = $node->getFirstToken();
```

Comment:
I'm not sure, but I think there's a scenario where this won't work: for example, if you want to ban function invocations to a particular function and make a linter of type `FunctionCallExpression`, then you won't have the preceding line here if the function call is in a program like
```
// HHAST_FIXME[NoCallsToBanMeLinter]
if (ban_me()) {
  return;
}
```
Not sure if it'd be better to walk `$parents` and look for the right comments or to use `$error->getPosition()` and look at surrounding lines. Or maybe something else?

Reply:
`if (/* HHAST_FIXME[NoCallsToBanMeLinter] */ ban_me()) { return; }` attaches the comment to the trail of `(` and not the leading of `ban_me()` 🤔

Reply:
I think we need to check:
 - own leading
 - leading of every node up to the first statement (or possibly expression)
 - trailing of the previous sibling

Reply:
@fredemmott could you help me understand this part a bit better? "leading of every node up to the first statement (or possibly expression)". I assume this means we walk up the parents until we hit a certain node? The actual IfStatement, ForStatement etc classes or something else?

Reply:
e.g. you probably shouldn't be able to use this to suppress every instance of an error in a function